core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -107,33 +115,6 @@
   @Override
   public void start(Listener<RespT> observer, Metadata headers) {
     Preconditions.checkState(stream == null, "Already started");
-    Long deadlineNanoTime = callOptions.getDeadlineNanoTime();
```

Comment:
In the previous version of the code, all the auto header changing and addition was done in a single place, and it was really easy to see where new code should be added (or where to look in case of a bug).  In the new version, TIMEOUT_KEY is now moved away from it, and there are two places.  

Consider moving the rest of the header processing code along with this, to keep it in one place.